Allow to generate categories

PascalBrouwers opened the pull request.

PascalBrouwers committed:
Allow to generate categories

eInyzant commented:
Thank you for your patch. I just used it and it worked like a charm.

gediminaskv commented:
I've got [Exception]
  Item (Magento\Catalog\Model\Category\Interceptor) with the same ID "57" alr
  eady exists.

colabsadmin commented:
Life saver!  Thank you!

andrewhowdencom committed:
Proxy the dependencies in the URL generation command

With Magento 2.1.8+ CLI commands are constructed when the CLI client is
invoked. Additionally, DI will construct any objects that those commands
type in their constructors, so on and so fourth down the stack.

The problem is, this extension may exist in an environment in which the
store is not yet installed, and the database is thus unavailable. A
common place this is used is in 2.1.x in CI, as the database is required
in order to generate static content.

This commit proxies the dependencies of this constructor, such that the
constructor can be used (and thus appear in the CLI menu) but not
invoked while the store is not installed.

PascalBrouwers committed:
Merge pull request #1 from sitewards/feature/generate_category

Proxy the dependencies in the URL generation command

peterjaap commented:
My fork also does this (along with url_path regeneration) although I use a slightly different approach; https://github.com/peterjaap/magento2-regenurl

PascalBrouwers committed:
Fix regenerating categories when category has no childs

PascalBrouwers committed:
Merge branch 'feature/generate_category' of github.com:PascalBrouwers/magento2-regenurl into feature/generate_category

PascalBrouwers commented:
Added some fixes when categories without childs would not be processed. @Iazel would be nice if this could be merged.

peterjaap commented:
@PascalBrouwers @Iazel hasn't been active on Github for months. My fork is the one with the fixes now. Maybe you want to do your PR on my fork?

PascalBrouwers commented:
Sure
